dialogue-core/src/main/java/com/palantir/dialogue/core/LimitedChannel.java

```diff
@@ -27,6 +28,6 @@
  * overloaded). Semantics match {@link com.palantir.dialogue.Channel} aside from returning an
  * {@link Optional optional response}.
  */
-interface LimitedChannel {
+interface LimitedChannel extends Channel {
```

Comment:
this seems really strange as a semantic; the idea of having a LimitedChannel and a Channel are sort of core to ensure separation of concerns; would love for a discussion around this that includes @ellisjoe